packages/react/src/Heading/getHeadingTag.ts

```diff
@@ -1,14 +1,3 @@
 import type { HeadingProps } from './Heading'
 
-export function getHeadingTag(level: HeadingProps['level']) {
-  switch (level) {
-    case 2:
-      return 'h2'
-    case 3:
-      return 'h3'
-    case 4:
-      return 'h4'
-    default:
-      return 'h1'
-  }
-}
+export const getHeadingTag = (level: HeadingProps['level']): `h${HeadingProps['level']}` => `h${level}`
```

Comment:
Without Typescript you can enter anything as `level`. The old function keeps working in that case, the new one doesn't.

Reply:
True, but the idea of this PR is that we can’t set a default. Only after I changed the `default` case to `case 1:` did I decide to refactor for simplicity.

What should this function return in JavaScript-land if a level of `5` of `a` was provided?

Reply:
Hm, tricky question. Do we want to allow invalid HTML or an incorrect heading structure? And which is more easily detected as erroneous?

[Utrecht uses a default of `h1`](https://github.com/nl-design-system/utrecht/blob/de1c443072f724f825d432b41d2aacefc83c2402/packages/component-library-react/src/Heading.tsx#L29) even though `level` is required, so they've opted to not allow invalid HTML. 

I think I prefer that route. Invalid HTML is almost completely unaccessible, an incorrect heading structure is still somewhat usable. Invalid HTML is easier to spot though.

Reply:
Yes, seems fair! I restored the function and added a comment.

Reply:
And allowed `null` as an argument to support that default value in Context.